### PR TITLE
Add --input argument

### PR DESCRIPTION
**Pull request by snease**
This argument accepts a path to a file. The data contained in the file will be passed to the command via stdin.

Closes sharkdp/hyperfine#541

### CODE REVIEW

File: src/options.rs [hunk 1]
```diff
@@ -196,8 +226,11 @@ pub struct Options {
     /// What to do with the output of the benchmarked command
     pub command_output_policy: CommandOutputPolicy,
 
-    /// Which time unit to use when displaying resuls
+    /// Which time unit to use when displaying results
     pub time_unit: Option<Unit>,
+
+    /// Where input to the benchmarked command comes from
+    pub command_input_policy: CommandInputPolicy,
```
Comment (sharkdp) [inline src/options.rs:233]: Maybe move this above `command_output_policy`?

File: src/benchmark/executor.rs [hunk 2]
```diff
@@ -1,7 +1,9 @@
 use std::process::{ExitStatus, Stdio};
```
Comment (sharkdp) [inline src/benchmark/executor.rs:1]: Can we please fie warning here (unused import: `Stdio`)

File: src/benchmark/executor.rs [hunk 3]
```diff
@@ -36,11 +38,13 @@ pub trait Executor {
 fn run_command_and_measure_common(
     mut command: std::process::Command,
     command_failure_action: CmdFailureAction,
+    command_input_policy: &CommandInputPolicy,
     command_output_policy: &CommandOutputPolicy,
     command_name: &str,
 ) -> Result<TimerResult> {
     let (stdout, stderr) = command_output_policy.get_stdout_stderr()?;
-    command.stdin(Stdio::null()).stdout(stdout).stderr(stderr);
+    let stdin = command_input_policy.get_stdin()?;
```
Comment (sharkdp) [inline src/benchmark/executor.rs:46]: move this above the `stdout, sterr` line to be consistent with the ordering elsewhere (input before output)?

File: src/cli.rs [hunk 4]
```diff
@@ -311,6 +311,14 @@ fn build_command() -> Command<'static> {
             .hide(true)
             .help("Enable debug mode which does not actually run commands, but returns fake times when the command is 'sleep <time>'.")
         )
+        .arg(
+            Arg::new("stdin-data")
```
Comment (sharkdp) [inline src/cli.rs:315]: Can we maybe rename this to `--input`, in analogy to `--output`? Or is there any reason against this?

Comment (sharkdp) [reply src/cli.rs:315]: And also change the option such that it works similar to `--output`. Maybe:
```
--input <FROM>
  Control where the input of the benchmark comes from. <FROM> can be:

  null: Read from /dev/null (the default).

  <FILE>: Read the input from the given file
```

File: src/options.rs [hunk 5]
```diff
@@ -110,6 +110,36 @@ impl Default for RunBounds {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum CommandInputPolicy {
+    /// Redirect from the null device
+    Null,
+
+    /// Redirect input from a file
```
Comment (sharkdp) [inline src/options.rs:118]:
```suggestion
    /// Read input from a file
```

File: src/options.rs [hunk 6]
```diff
@@ -110,6 +110,36 @@ impl Default for RunBounds {
     }
 }
 
+#[derive(Debug, Clone, PartialEq)]
+pub enum CommandInputPolicy {
+    /// Redirect from the null device
```
Comment (sharkdp) [inline src/options.rs:115]:
```suggestion
    /// Read from the null device
```

**Review by sharkdp — CHANGES_REQUESTED** [review]
Thank you very much for your contribution!

**Comment by snease** [comment]
Happy to help! Thanks for creating a great project.

File: tests/example_stdin_data [hunk 7]
```diff
@@ -0,0 +1 @@
+This data is passed to the command via stdin
```
Comment (sharkdp) [inline tests/example_stdin_data:1]: Minor: "stdin" is usually input from the terminal. While this is input from a file. This is a bit confusing when reading the test below. Can we maybe change this to
```suggestion
This text is part of a file
```

File: tests/integration_tests.rs [hunk 8]
```diff
@@ -215,6 +215,34 @@ fn runs_commands_using_user_defined_shell() {
...
+        .arg("--runs=1")
+        .arg("--input=example_stdin_data")
+        .arg("--show-output")
+        .arg("cat")
+        .assert()
+        .success()
+        .stdout(predicate::str::contains(
+            "This data is passed to the command via stdin",
+        ));
+}
```
Comment (sharkdp) [inline tests/integration_tests.rs:230]:
```suggestion
fn can_pass_input_to_command_from_a_file() {
    hyperfine()
        .arg("--runs=1")
        .arg("--input=example_input_file.txt")
        .arg("--show-output")
        .arg("cat")
        .assert()
        .success()
        .stdout(predicate::str::contains(
            "This text is part of a file",
        ));
}
```

File: tests/integration_tests.rs [hunk 9]
```diff
@@ -215,6 +215,34 @@ fn runs_commands_using_user_defined_shell() {
...
+
+#[test]
+fn fails_if_invalid_stdin_data_file_provided() {
+    hyperfine()
+        .arg("--runs=1")
+        .arg("--input=example_stdin_data_invalid")
```
Comment (sharkdp) [inline tests/integration_tests.rs:236]:
```suggestion
        .arg("--input=example_non_existent_file.txt")
```

File: src/benchmark/executor.rs (hunk 3 above)
Comment (sharkdp) [reply src/benchmark/executor.rs:46]: That has not been resolved yet.

**Review by sharkdp — APPROVED** [review]

**Pull request merged**